mypyc/lib-rt/misc_ops.c

```diff
@@ -643,3 +643,46 @@ CPy_Super(PyObject *builtins, PyObject *self) {
     Py_DECREF(super_type);
     return result;
 }
+
+// The following code is a simplification of cpython/ceval.c/import_from()
+PyObject *CPyImport_ImportFrom(PyObject *module, PyObject *package_name,
+                               PyObject *import_name, PyObject *as_name) {
+
+    PyObject *x;
+    // check if the imported module has an attribute by that name
+    if (_PyObject_LookupAttr(module, import_name, &x) == 0) {
+        // if not, attempt to import a submodule with that name
+        PyObject *fullmodname = PyUnicode_FromFormat("%U.%U", package_name, import_name);
+        if (fullmodname == NULL) {
+            goto fail;
+        }
+
+        // The following code is a simplification of cpython/import.c/PyImport_GetModule()
+        Py_INCREF(module);
```

Comment:
This incref and the decref below seem redundant, since the caller will continue to hold a reference to these.